### ISSUE TITLE
GraphSCI implementation

### ISSUE THREAD

**Issue by GlancerZ** [OmicsML/dance#17]
For the imputation model GraphSCI, I think the AE part is wrong, the size_factor is cell_specific, so it should have the same number as cells, but in your implementation the size_factor number is gene-specific. For the inputs of AE, it should be the transpose of the input of GNN, so the input's shape should be cell*gene.

**Comment by WenzhuoTang** [comment]
Hi @GlancerZ , The original AE of GraphSCI takes input as cell*gene and the GCN module takes adjacency matrix of dimension num_gene*num_gene. If we keep that original setting, the trained model will only accept testing set with exactly the same dimensions of training set. So we transpose the input of AE to accommodate the train-test setting.

**Comment by WenzhuoTang** [comment]
Hi @GlancerZ , The original AE of GraphSCI takes input as cell\*gene and the GCN module takes adjacency matrix of dimension num_gene\*num_gene. If we keep that original setting, the trained model will only accept testing set with exactly the same dimensions of training set. So we transpose the input of AE to accommodate the train-test setting.

**Comment by GlancerZ** [comment]
I see, you use 80% gene and all gene to generate graph, so you use num_gene*num_cell for AE input, that's really help the test. But there is another problem, could we compute the zinb likelihood loss on gene dim, because usually we model one cell on zinb distribution.

**Comment by WenzhuoTang** [comment]
That is a good point. Currently the package is still under development. We may fix it in the next version. Thank you for the very helpful comment!

**Comment by GlancerZ** [comment]
Thanks for your reply, your work is really excellent！

**Comment by RemyLau** [comment]
Closing this issue as it has been stale for three months. Feel free to comment and reopen if there are any remaining issues related to this thread.
